static/js/popover_menus.js

```diff
@@ -85,4 +150,80 @@ export function initialize() {
             compose_mobile_button_popover_displayed = false;
         },
     });
+
+    function schedule_message_to_custom_date(send_at_time) {
+        if (!send_at_time) {
+            return;
+        }
+
+        const request = compose.create_message_object();
+        request.content = `/schedule ${send_at_time}\n` + request.content;
+        reminder.schedule_message(request);
+    }
+
+    function set_compose_box_schedule(element) {
+        const send_later_in = $(element).attr("id");
+        if ($(element).attr("class") === "send_later_hours") {
+            const send_at_time = format(
+                add(new Date(), {minutes: send_later_hours[send_later_in].hours * 60}),
+                "MMM d yyyy h:mm a",
+            );
+            return send_at_time;
+        } else if ($(element).attr("class") === "send_later_tomorrow") {
+            const send_time = send_later_tomorrow[send_later_in].time;
+            const date = new Date();
+            const scheduled_date = date.setDate(date.getDate() + 1);
+            const send_at_time = format(scheduled_date, "MMM d yyyy ") + send_time;
+            return send_at_time;
+        } else if ($(element).attr("class") === "send_later_days_and_weeks") {
+            const send_at_time = format(
+                add(new Date(), {days: send_later_days_and_weeks[send_later_in].days}),
+                "MMM d yyyy h:mm a",
+            );
+            return send_at_time;
+        }
```

Comment:
Hmm, why do we need these conditionals?  I wonder if we could instead just have the `send_later_options` object have the values and display strings.

Reply:
I simplified it a bit but we still need those conditions since everyone needs to be handled separately.